llvm/include/llvm/CodeGen/MachineDominators.h

```diff
@@ -266,6 +266,7 @@ class MachineDominatorTree : public DomTreeBase<MachineBasicBlock> {
 /// \brief Analysis pass which computes a \c MachineDominatorTree.
 class MachineDominatorTreeWrapperPass : public MachineFunctionPass {
   MachineDominatorTree DT;
+  bool IsDomTreeEmpty = true;
```

Comment:
Why is this necessary?

Reply:
If a pass preserves machine dominator tree analysis and machine dominator tree analysis result doesn't exists, `verifyAnalysis` will verify an empty dominator tree, which will trigger assertion. In old implementation it is a `unique_ptr` and can be `nullptr`.

Reply:
Either make it std::optional or fix the verifier to not assert on the empty case